api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -286,6 +287,13 @@ type ProtectedPVC struct {
 	LastSyncBytes *int64 `json:"lastSyncBytes,omitempty"`
 }
 
+func (p *ProtectedPVC) NamespacedName() types.NamespacedName {
```

Comment:
I would keep this file clean from implementation and put this function in the implementation file instead of this file?

Reply:
I don't think that can be done as ProtectedPVC belongs to the api package and all the implementations are in the controllers package?

However, I can change the 8 occurences I have to this call to build the NamespacedName instead if you would like to keep API package implementation free.

Something like this
```
	return types.NamespacedName{
		Namespace: p.Namespace,
		Name:      p.Name,
	}
	
```

Reply:
I didn't mean that. I meant it to be a free function. You put it in the `misc.go` or some other file like this:
```
func  NamespacedName(p *ProtectedPVC) types.NamespacedName {
	return types.NamespacedName{
		Namespace: p.Namespace,
		Name:      p.Name,
	}
}
```
So instead of:
`protectePVC.NamespacedName()`
it would be
`NamespacedName(protectedPVC)`

Reply:
Or even better, something like this:
```
func NamespacedName(o client.Object) types.NamespacedName {
	return types.NamespacedName{
		Namespace: o.GetName(),
		Name:      o.GetNamespace(),
	}
}
```

Reply:
Done